### ISSUE TITLE
standardise the bcl2fastq demultiplexing

### ISSUE THREAD

**Issue by ChrOertlin** [Clinical-Genomics/cg#2504]
## Description

In issue: https://github.com/Clinical-Genomics/cg/issues/2492 we saw that the way we currently write sample sheets for bcl2fastq causes unexpected issues with the fastq file names.

From the refinement we decided to standardise the workflow and that older and or legacy flow cells will require manual intervention for adhere to the new demultiplex flow.

As part of this we started a bcl2fastq demultiplexing run with the following command:

`singularity exec --bind /home/proj/stage/test_demux,/home/proj/stage/flow_cells/,/home/proj/stage/flow_cells/'$SLURM_JOB_ID':/run/user/$(id -u) /home/proj/production/demux-on-hasta/novaseq/container/bcl2fastq_v2-20-0.sif bcl2fastq --loading-threads 3 --processing-threads 15 --writing-threads 3 --runfolder-dir /home/proj/stage/flow_cells/221019_A00689_0663_BXXXXXXX --output-dir /home/proj/stage/test_demux/221019_A00689_0663_BXXXXXXX --sample-sheet /home/proj/stage/flow_cells/221019_A00689_0663_BXXXXXXX/SampleSheet.csv --barcode-mismatches 1`

Note: the `--output-dir` does not contain the `Unaligned` directory. The 'SampleSheet.csv' file does not contain a `Sample_Name` column.
 

This run produced the following output structure:

`221019_A00689_0663_BXXXXXXX/
          <Project>/
           Reports/
            Stats/
`
Under the `<Project>` folder all fastqs were stored as `<sample_internal_id>_<rest_of_fastq_file>`. This shows the wanted behaviour of bcl2fastq when omitting the `Sample_Name` column from the sample sheet.

In addition, we could potentially eliminate the `Unaligned` folder from the demultiplexing flow. As this folder is not a requirement any longer at least to my knowldge.

Thus, in addition to changes outlined in: https://github.com/Clinical-Genomics/cg/issues/2492 we can remove the `Unaligned` folder from the `bcl2fastq` sbatch command. 

This will then reduce the amount of patterns required to pick up fastq files for our post-processing automation flow to.

```Python

 # The flat output structure for flow cells demultiplex with BCLConvert on hasta
    root_pattern = f"{sample_internal_id}_S*_L*_R*_*{FileExtensions.FASTQ}{FileExtensions.GZIP}"

    # The default structure for bcl2fastq
    bcl2fastq_pattern = (
        f"*/{sample_internal_id}_*{FileExtensions.FASTQ}{FileExtensions.GZIP}"
    )

    # The pattern for novaseqx flow cells demultiplexed on board of the sequencer
    demux_on_sequencer_pattern = (
        f"BCLConvert/fastq/{sample_internal_id}"
        f"_S*_L*_R*_*{FileExtensions.FASTQ}{FileExtensions.GZIP}"
    )
```

## Suggested solution

*Write here if there is a suggested way to solve the issue*

## This can be closed when

*Describe what needs to be done for this issue to be closed*

## Blocked by

*If there are any blocking issues/prs/things in this or other repos. Please link to them.*

**Comment by Vince-janv** [comment]
### Refinement discussion:
Sounds good, try also removing `Project` column alternately pass an option to the CLI

See https://github.com/Clinical-Genomics/cg/issues/2492#issuecomment-1737377109 for more info

**Comment by ChrOertlin** [comment]
![image](https://github.com/Clinical-Genomics/cg/assets/8860786/e3adc412-a3a1-4f4b-9553-ca26b5203bf8)

**Comment by ChrOertlin** [comment]
Removing the "Project" column gives indeed a flat structure as is the case with the BCL Convert Demultiplexing flow.

**Comment by seallard** [comment]
I have some vague memory on us settling on removing support for bcl2fastq, maybe we can close this?

**Comment by Vince-janv** [comment]
> I have some vague memory of us settling on removing support for bcl2fastq, maybe we can close this?

Yes! Closing in favour of https://github.com/Clinical-Genomics/project-planning/issues/521
